iothub/device/src/Exceptions/IotHubClientException.cs

```diff
@@ -19,25 +20,43 @@ public class IotHubClientException : Exception
         [NonSerialized]
         private const string TrackingIdValueSerializationStoreName = "IotHubClientException-TrackingId";
 
-        private HashSet<IotHubStatusCode> transientStatusCodes = new HashSet<IotHubStatusCode>
+        private HashSet<IotHubClientErrorCode> transientErrorCodes = new HashSet<IotHubClientErrorCode>
         {
-            IotHubStatusCode.QuotaExceeded,
-            IotHubStatusCode.ServerError,
-            IotHubStatusCode.ServerBusy,
-            IotHubStatusCode.Throttled,
-            IotHubStatusCode.Timeout,
-            IotHubStatusCode.NetworkErrors,
+            IotHubClientErrorCode.QuotaExceeded,
+            IotHubClientErrorCode.ServerError,
+            IotHubClientErrorCode.ServerBusy,
+            IotHubClientErrorCode.Throttled,
+            IotHubClientErrorCode.Timeout,
+            IotHubClientErrorCode.NetworkErrors,
+        };
+
+        private Dictionary<IotHubClientErrorCode, HttpStatusCode> httpStatusCodes = new Dictionary<IotHubClientErrorCode, HttpStatusCode>()
```

Comment:
Taking the best effort to map from ErrorCode to StatusCode

Reply:
```suggestion
        private Dictionary<IotHubClientErrorCode, HttpStatusCode> httpStatusCodes = new()
```